### PR TITLE
Support batch normalization

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/layers.py [hunk 1]
```diff
@@ -336,6 +352,7 @@ def __init__(self,
             output_size (int): output size
             n (int): n independent ``FC`` layers
             activation (torch.nn.functional):
+            use_bn (bool): whehter use Batch Normalization.
```
Comment (Haichao-Zhang) [inline alf/layers.py]: whehter -> whether

File: alf/networks/network.py [hunk 2]
```diff
@@ -188,8 +188,11 @@ def output_spec(self):
         can be obtained easily in other ways.
         """
         if self._output_spec is None:
+            training = self.training
+            self.eval()
```
Comment (Haichao-Zhang) [inline alf/networks/network.py:192]: Similar switch between ```module.train()``` and ```module.eval()``` mode might be needed once BN is introduced (e.g. in rollout, play..) ?

Comment (emailweixu) [reply alf/networks/network.py:192]: Done. Though I am not very satisfied how eval and train are implemented. They are using recursive calls to go through all submodules. It's kind of expensive. But it seems there no good way given how pytorch works.

File: alf/layers.py (hunk 1 above)
Comment (emailweixu) [reply alf/layers.py]: fixed

**Review by Haichao-Zhang — APPROVED** [review]

File: alf/networks/network.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/networks/network.py:192]: Right. This will add some extra cost. We can try to think of ways to improve it in the future.
